### ISSUE TITLE
Feature Request: Debugger 

### ISSUE THREAD

**Issue by rachtibat** [chr5tphr/zennit#187]
Hey,

I think it would be really nice, if we could check if the composite actually attached to all the modules we named and to see if there are any modules left that have no rules attached to.
In the long run - but much more difficult to realize - it would be nice to see, if there are any non-Module elements that unexpectedly changed the Relevance values.

Best

**Comment by rachtibat** [comment]
Regarding the second feature "detecting unexpectedly changed relevance values", this could be implemented by introducing a new rule/hook that 
1. returns fixed relevance values, e.g. torch.ones_like(input.shape) * 42.
2. and then checks at the input whether the gradient_inputs == our fixed values.

Best

**Comment by chr5tphr** [comment]
Hey Reduan,

yes, I think checking whether the composite assigned the rules correctly is a nice feature to have, as this is also done inside the tests. One sanity check that I also would like to introduce is #147, which is somewhat related.

Checking whether the gradient was modified in between rules may be a little challenging for anything that is not feed-forward, as the gradients may not be passed in the same order as they are defined within the model, but there might be a an elegant way to do it.
